core/src/execution/datafusion/expressions/negative.rs

```diff
@@ -0,0 +1,408 @@
+                                        from_type: "float".to_string(),
+                                    }
+                                    .into());
+                                }
+                            }
+                        }
+                        DataType::Float32 => {
+                            let float_array = array
+                                .as_any()
+                                .downcast_ref::<arrow::array::Float32Array>()
+                                .unwrap();
+                            for i in 0..float_array.len() {
+                                if float_array.value(i) <= f32::MIN
+                                    || float_array.value(i) >= f32::MAX
+                                {
+                                    return Err(CometError::ArithmeticOverflow {
+                                        from_type: "float".to_string(),
+                                    }
+                                    .into());
+                                }
+                            }
+                        }
+                        DataType::Float64 => {
+                            let float_array = array
+                                .as_any()
+                                .downcast_ref::<arrow::array::Float64Array>()
+                                .unwrap();
+                            for i in 0..float_array.len() {
+                                if float_array.value(i) <= f64::MIN
+                                    || float_array.value(i) >= f64::MAX
+                                {
+                                    return Err(CometError::ArithmeticOverflow {
+                                        from_type: "float".to_string(),
+                                    }
+                                    .into());
+                                }
+                            }
+                        }
+                        DataType::Interval(value) => {
+                            // Downcast the array to the appropriate interval type
+                            match value {
+                                arrow::datatypes::IntervalUnit::YearMonth => {
+                                    let interval_array = array
+                                        .as_any()
+                                        .downcast_ref::<arrow::array::IntervalYearMonthArray>()
+                                        .unwrap();
+                                    for i in 0..interval_array.len() {
+                                        if interval_array.value(i) == i32::MIN
+                                            || interval_array.value(i) == i32::MAX
+                                        {
+                                            return Err(CometError::ArithmeticOverflow {
+                                                from_type: "interval".to_string(),
+                                            }
```

Comment:
nit: consider adding a function to create the `CometError::ArithmeticOverflow` err to reduce boilerplate of creating the struct.